### PR TITLE
[AppConfig-PerfTest] init appconfiguraiton performance test

### PR DESCRIPTION
**Pull request by mssfang**
- Init the an perf pacakges, fixes: https://github.com/Azure/azure-sdk-for-java/issues/19452, https://github.com/Azure/azure-sdk-for-java/issues/14311
- List 20 configuration settings (which is the main user scenarios)

```
λ java -jar sdk\appconfiguration\azure-data-appconfiguration-perf\target\azure-data-appconfiguration-pef-1.0.0-beta.1-jar-with-dependencies.jar listconfigurationsettings --warmup 1 --iterations 1 --parallel 6 --duration 10 --count 20
=== Options ===
{
  "count" : 20,
  "duration" : 10,
  "host" : null,
  "insecure" : false,
  "iterations" : 1,
  "noCleanup" : false,
  "parallel" : 6,
  "port" : -1,
  "size" : 10240,
  "sync" : false,
  "warmup" : 1
}

=== Setup ===
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
..

=== Warmup ===
Current         Total           Average
86              86              87.81
86              86              87.81

=== Results ===
Completed 86 operations in a weighted-average of 0.98s (87.81 ops/s, 0.011 s/op)

=== Test ===
Current         Total           Average
101             101             102.83
125             226             114.75
132             358             120.74
127             485             121.79
127             612             122.87
129             741             123.77
123             864             123.82
118             982             122.80
131             1113            123.83
130             1243            124.60
0               1243            124.60

=== Results ===
Completed 1,243 operations in a weighted-average of 9.98s (124.60 ops/s, 0.008 s/op)

=== Cleanup ===
.

```

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration-perf/src/main/java/com/azure/data/appconfiguration/perf/ListConfigurationSettingsTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+        return Flux.concat(settingMonoList).then();
+    }
+
+    @Override
+    public Mono<Void> globalCleanupAsync() {
+        List<Mono<ConfigurationSetting>> settingMonoList = new ArrayList<>();
+        for (int i = 0; i < settingCount; i++) {
+            settingMonoList.add(configurationAsyncClient.deleteConfigurationSetting(
+                new ConfigurationSetting().setKey(KEY_PREFIX + i).setValue(SETTING_VALUE)));
```
Comment (pakrym) [inline sdk/appconfiguration/azure-data-appconfiguration-perf/src/main/java/com/azure/data/appconfiguration/perf/ListConfigurationSettingsTest.java:52]: Don't you have to delete the key here?

Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration-perf/src/main/java/com/azure/data/appconfiguration/perf/ListConfigurationSettingsTest.java:52]: deleteConfigurationSetting() deletes the setting in the resource

**Comment by pakrym** [comment]
LGTM

File: pom.xml [hunk 2]
```diff
@@ -778,4 +778,7 @@
       </modules>
     </profile>
   </profiles>
+    <modules>
+        <module>sdk/appconfiguration/azure-data-appconfiguration-pef</module>
+    </modules>
```
Comment (alzimmermsft) [inline pom.xml:783]: This should be removed

File: sdk/appconfiguration/azure-data-appconfiguration-perf/src/main/java/com/azure/data/appconfiguration/perf/ListConfigurationSettingsTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.List;
+
+/**
+ * Performs listing multiple configuration settings.
+ */
+public class ListConfigurationSettingsTest extends ServiceTest<PerfStressOptions> {
+    private static final String KEY_PREFIX = "keyPrefix";
+    private static final String SETTING_VALUE = "value";
+
+    private static int settingCount = 1;
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration-perf/src/main/java/com/azure/data/appconfiguration/perf/ListConfigurationSettingsTest.java:23]: Should make this into a configuration value. 

Also, is 1 configuration setting in listing an appropriate default? Does 1 seem like the most common listing size?

Comment (pakrym) [reply sdk/appconfiguration/azure-data-appconfiguration-perf/src/main/java/com/azure/data/appconfiguration/perf/ListConfigurationSettingsTest.java:23]: .NET has this build-in options type https://github.com/Azure/azure-sdk-for-net/blob/master/common/Perf/Azure.Test.Perf/CountOptions.cs#L10

with count == 10 by-default.

File: pom.xml (hunk 2 above)
Comment (mssfang) [reply pom.xml:783]: Missed this one. Thanks Alan

File: sdk/appconfiguration/azure-data-appconfiguration-perf/src/main/java/com/azure/data/appconfiguration/perf/ListConfigurationSettingsTest.java (hunk 3 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration-perf/src/main/java/com/azure/data/appconfiguration/perf/ListConfigurationSettingsTest.java:23]: @pakrym we need to have 20 settings, is it?

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**